File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/02e969828fa8c90ecf63941da505769c.go
package main

import (
	"fmt"
	"math"
	"time"
)

var TimeStart time.Time
var TimeEnd time.Time

const Mod int64 = 1000000007
const INF int64 = 1e9
const LINF int64 = 1e18
const Pi float64 = 3.141592653589793116
const EPS float64 = 1e-9
const Gold float64 = (1 + math.Sqrt(5)) / 2

var keymod = []int64{1000000007, 1000000009, 1000000021, 1000000033}
var keyCount = len(keymod)

func modPow(a, b, M int64) int64 {
	if b == 0 {
		return 1
	}
	tmp := modPow(a, b/2, M)
	if b%2 == 0 {
		return (tmp * tmp) % M
	}
	return (((tmp * tmp) % M) * a) % M
}

func Input() {
	var n, m int64
	fmt.Scan(&n, &m)
}

func Solve() {
	var n, m int64
	A := (modPow(3, n, m) + m - 1) % m
	fmt.Println(A)
}

func main() {
	ControlIO()
	Input()
	TimerStart()
	Solve()
	TimerStop()
}

func ControlIO() {
	// Add your code here for any specific IO control
}

func TimerStart() {
	TimeStart = time.Now()
}

func TimerStop() {
	TimeEnd = time.Now()
	ElapsedTime := TimeEnd.Sub(TimeStart)
	fmt.Printf("\n\nTime elapsed: %f seconds.\n", ElapsedTime.Seconds())
}

func Exit() {
	TimerStop()
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/C/02e969828fa8c90ecf63941da505769c.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>

#define MOD 1000000007LL
#define INF 1e9
#define LINF 1e18
#define EPS 1e-9

typedef long long i64;

i64 n, m;

i64 modPow(i64 a, i64 b, i64 M) {
    if (b == 0) return 1;
    i64 tmp = modPow(a, b / 2, M);
    if (b % 2 == 0) return (tmp * tmp) % M;
    return (((tmp * tmp) % M) * a) % M;
}

void Input() {
    scanf("%lld %lld", &n, &m);
}

void Solve() {
    i64 A = (modPow(3, n, m) + m - 1) % m;
    printf("%lld", A);
}

void ControlIO(int argc, char* argv[]) {
    if (argc > 1) {
        freopen(argv[1], "r", stdin);
    }
    if (argc > 2) {
        freopen(argv[2], "w", stdout);
    }
}

void TimerStart(struct timespec *start) {
    clock_gettime(CLOCK_MONOTONIC, start);
}

void TimerStop(struct timespec *start) {
    struct timespec end;
    clock_gettime(CLOCK_MONOTONIC, &end);
    double elapsed = (end.tv_sec - start->tv_sec) + 
                    (end.tv_nsec - start->tv_nsec) / 1e9;
    printf("\n\nTime elapsed: %f seconds.\n", elapsed);
}

int main(int argc, char* argv[]) {
    struct timespec TimeStart;
    ControlIO(argc, argv);
    TimerStart(&TimeStart);
    
    Input();
    Solve();
    
    TimerStop(&TimeStart);
    return 0;
}
